### PR TITLE
Merged GSFX into the official Gamebuino META API

### PR DESCRIPTION
**Pull request by EspeuteClement**

### CODE REVIEW

File: libraries/Gamebuino-Meta/src/utility/Sound/Sound.cpp [hunk 1]
```diff
@@ -211,6 +214,28 @@ int8_t Sound::play(Sound_Handler* handler, bool loop) {
 #endif // SOUND_CHANNELS
 }
 
+void init_fx_channel(int channel_id) {
+	if (fx_channels[channel_id].handler == nullptr){
+		fx_channels[channel_id].size = 2048 * SOUND_FREQ / 44100;
```
Comment (Sorunome) [inline libraries/Gamebuino-Meta/src/utility/Sound/Sound.cpp]: Please use `SOUND_BUFFERSIZE` instead

File: libraries/Gamebuino-Meta/src/utility/Sound/Sound.cpp [hunk 2]
```diff
@@ -262,6 +287,10 @@ void Sound::update() {
 			handlers[i] = 0;
 		}
 	}
+
+	if (fx_channels[0].handler) {
```
Comment (Sorunome) [inline libraries/Gamebuino-Meta/src/utility/Sound/Sound.cpp]: if you have a configurable amount of fx_channels, you should iterate over them. However, I do not see a case where you would need multiple, so why make it an array at all?

File: libraries/Gamebuino-Meta/src/utility/Sound/Sound_FX.cpp [hunk 3]
```diff
@@ -0,0 +1,154 @@
+/*
+This file is part of the Gamebuino-Meta library,
+Copyright (c) Aadalie 2017
```
Comment (Sorunome) [inline libraries/Gamebuino-Meta/src/utility/Sound/Sound_FX.cpp]: it's 2018! Also, it says (c) Aadalie, which is the gamebuino company, if you want to give over copyright, sure, it is up to you, though

File: libraries/Gamebuino-Meta/src/utility/Sound/Sound_FX.cpp [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+If not, see <http://www.gnu.org/licenses/>.
+
+Authors:
+- Valden
+*/
+
+#include "Sound_FX.h"
+#include <Gamebuino-Meta.h>
+
+namespace Gamebuino_Meta {
```
Comment (Sorunome) [inline libraries/Gamebuino-Meta/src/utility/Sound/Sound_FX.cpp]: Please don't indent inside the namespace, instead leave a few extra blank lines

File: libraries/Gamebuino-Meta/src/utility/Sound/Sound_FX.cpp [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+			volume =  _square_polarity * volume;
+			uint8_t v = (uint8_t)volume;
+			uint32_t sample = (v << 24) | (v << 16) | (v << 8) | v;
+
+			_head_index = (_head_index + 1) % (parent_channel->size >> 2);
+			_current_Sound_FX_time += 4 * SR_DIVIDER;
+			((uint32_t*)(&parent_channel->buffer[0]))[_head_index] = sample;
+		} while (_head_index != target >> 2);
+	}
+}
```
Comment (Sorunome) [inline libraries/Gamebuino-Meta/src/utility/Sound/Sound_FX.cpp]: `// namespace Gamebuino_Meta`

File: libraries/Gamebuino-Meta/src/utility/Sound/Sound_FX.h [hunk 6]
```diff
@@ -0,0 +1,85 @@
+/*
+This file is part of the Gamebuino-Meta library,
+Copyright (c) Aadalie 2017
```
Comment (Sorunome) [inline libraries/Gamebuino-Meta/src/utility/Sound/Sound_FX.h]: Same stuff as with the cpp file

File: libraries/Gamebuino-Meta/src/utility/Sound/Sound.h [hunk 7]
```diff
@@ -77,6 +113,10 @@ class Sound {
 	};
 	
 	int8_t play(Sound_Handler* handler, bool loop = false);
+
+	void fx(const Sound_FX & fx);
```
Comment (Sorunome) [inline libraries/Gamebuino-Meta/src/utility/Sound/Sound.h]: Auto-detect length via templates?

Comment (EspeuteClement) [reply libraries/Gamebuino-Meta/src/utility/Sound/Sound.h]: The sound FX structure has a hidden flag that says if the fx should continue or not, like this
```
Gamebuino_Meta::Sound_FX sfx_test[8] =
{
	{ Gamebuino_Meta::Sound_FX_Wave::SQUARE_CONTINUE, 19968,0,8192,0,1000 },
	{ Gamebuino_Meta::Sound_FX_Wave::SQUARE, 19968,-5,4096,0,10000 },
};
```

Comment (Sorunome) [reply libraries/Gamebuino-Meta/src/utility/Sound/Sound.h]: Ah yes, that will do it, too

**Pull request merged**